src/main/java/org/alliancegenome/curation_api/services/mati/MaTIService.java

```diff
@@ -0,0 +1,79 @@
+package org.alliancegenome.curation_api.services.mati;
+
+import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
+import org.alliancegenome.curation_api.model.entities.IdentifiersRange;
+
+import javax.enterprise.context.ApplicationScoped;
+import java.io.IOException;
+import java.util.Base64;
+
+@ApplicationScoped
+public class MaTIService {
+//	@ConfigProperty(name = "okta.client.id") TODO: reading @ConfigProperties is broken inside integ tests
+	private String client_id;
+//	@ConfigProperty(name = "okta.client.secret") TODO: workaround with env vars needs github actions support
+	private  String client_secret;
+//	@ConfigProperty(name = "okta.url")
+	private String okta_url;
+//	@ConfigProperty(name = "okta.scopes")
+	private String okta_scopes;
+//	@ConfigProperty(name = "mati.url")
+	private String mati_url;
+
+	public MaTIService() {
+		client_id = System.getenv("OKTA_CLIENT_ID");
+		client_secret = System.getenv("OKTA_CLIENT_SECRET");
+		okta_url = System.getenv("OKTA_URL");
+		okta_scopes = System.getenv("OKTA_SCOPES");
+		mati_url = System.getenv("MATI_URL");
```

Comment:
These should be pulled via the @ConfigProperty and some default values can be added to the application.properties.defaults file. See the Bulk services for examples using a `Instance<String>`  doesn't throw the error when the variables are not defined. You can see an example in the file: https://github.com/alliance-genome/agr_curation/blob/alpha/src/main/java/org/alliancegenome/curation_api/auth/AuthenticationFilter.java

Reply:
From a class with  @ConfigProperty, the creation of an object in an integration test yields NULL. It does not matter if the attributes are typed directly or with Instance. 

The use of ENV vars is the alternative I found, is there another?

Reply:
with integration tests, those values just need to be set otherwise yes they will be NULL... and we can make sure and have them set when running.

Reply:
Here is the documentation on how @ConfigProperty works.

https://quarkus.io/guides/config-reference#configuration-sources

As you can see there is a chain of configuration options which System.getenv is included. So it's better to use the convention. If the values are not coming through its not he @ConfigProperty that is the issue, its that they are not being set correctly.

Reply:
On another note, MATI_URL is not going to be available to github action. Mati is behind the AWS firewall. @mluypaert this is another reason I wanted to make the Mati API public, it is protected via okta auth, and should be available for this kind of thing.

Reply:
Makint it public also eases development and testing for Blue Team

Reply:
Last commit cleans up the comments and the integration test runs with all the others.